presto-main/src/main/java/io/prestosql/sql/analyzer/Analysis.java

```diff
@@ -222,8 +224,8 @@ public void resetUpdateType()
 
     public boolean isDeleteTarget(Table table)
     {
-        QualifiedObjectName name = tables.get(NodeRef.of(table)).getName();
-        return "DELETE".equals(updateType) && Optional.of(name).equals(target);
+        NodeRef<Table> tableRef = NodeRef.of(table);
+        return "DELETE".equals(updateType) && Optional.of(tableRef).equals(target.flatMap(UpdateTarget::getTable));
```

Comment:
the `target` should be set at this point and should contain a table, so

```
return "DELETE".equals(updateType) &&
  target.orElseThrow(...)
     .getTable().orElseThrow(...) == table; // intentional comparison by reference
```

Reply:
i'm aware this would broaden the scope of the change, but i think it would be a good change